dimod/core/structured.py

```diff
@@ -144,7 +148,7 @@ def to_networkx_graph(self):
 
         return G
 
-    def valid_bqm_graph(self, bqm: "BinaryQuadraticModel"):
+    def valid_bqm_graph(self, bqm: "BinaryQuadraticModel") -> bool:
```

Comment:
Think this doesn't need the quotes

Reply:
I put this in quotes in https://github.com/dwavesystems/dimod/pull/1101/commits/e6226b17a90d1ffefcd6593804ce68a61c14e24f because importing `BinaryQuadraticModel` to this file causes a circular import failure

Reply:
Ah, in that case you need to do
```python
if typing.TYPE_CHECKING:
    from dimod import BinaryQuadraticModel
```
another option would be to just
```python
import dimod
```
at the top and then do
```
def valid_bqm_graph(self, bqm: dimod.BinaryQuadraticModel) -> bool:
```

Reply:
Both alternatives still cause failures for the `import dimod` line in `conf.py` during doc build. 
```
def valid_bqm_graph(self, bqm: dimod.BinaryQuadraticModel) -> bool:
AttributeError: partially initialized module 'dimod' has no attribute 'BinaryQuadraticModel' (most likely due to a circular import)
```

Reply:
Typing is such a mess. Does 
```python
if typing.TYPE_CHECKING:
    from dimod import BinaryQuadraticModel
```
with
```python
def valid_bqm_graph(self, bqm: BinaryQuadraticModel) -> bool:
```
also raise the error? That surprises me

Reply:
I think that happens in the runtime: 
```
def valid_bqm_graph(self, bqm: BinaryQuadraticModel) -> bool:
NameError: name 'BinaryQuadraticModel' is not defined
```

Reply:
Right, this is where you need the `from __future__ import annotations`.

Reply:
Right! Just restored it and now builds. Thanks!